remoteconfig/client.go

```diff
@@ -0,0 +1,223 @@
+package remoteconfig
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"firebase.google.com/go/v4/internal"
+)
+
+const (
+	defaultRemoteConfigEndpoint = "https://firebaseremoteconfig.googleapis.com/v1"
+)
+
+// Client .
+type Client struct {
+	hc        *internal.HTTPClient
+	projectID string
+}
+
+func (c *Client) getRootURL() string {
+	return fmt.Sprintf("%s/projects/%s/remoteConfig", defaultRemoteConfigEndpoint, c.projectID)
+}
+
+// GetRemoteConfig https://firebase.google.com/docs/reference/remote-config/rest/v1/projects/getRemoteConfig
+func (c *Client) GetRemoteConfig(versionNumber string) (*Response, error) {
```

Comment:
Instead of this function we should have the following as mentioned in the proposed API.
```
func (c *Client) GetTemplate(ctx context.Context) (*Template, error)
func (c *Client) GetTemplateAtVersion(ctx context.Context, versionNumber string) (*Template, error)
```
You could simply change this to `func (c *Client) GetTemplate(ctx context.Context) (*Template, error)` if you don't plan to implement both. We can add `GetTemplateAtVersion()` in a future PR.

Reply:
Reconsidered and this looks like a good first step. It is fine to leave this as it is. No need to make further changes. Please ignore my comment.